chainer/functions/connection/linear.py

```diff
@@ -69,6 +69,81 @@ def backward(self, indexes, grad_outputs):
         return ret
 
 
+class LinearGradData(function_node.FunctionNode):
+
+    def __init__(self, linear):
+        return
+
+    def forward(self, inputs):
+        self.retain_inputs((0, 1))
+        gy, W = inputs
+
+        if not type_check.same_types(*inputs):
```

Comment:
We now implemented this check in FunctionNode-level (see https://github.com/chainer/chainer/pull/4029), so you don't have to check this.
(same for `LinearGradWeight`)

Reply:
I will remove it, thanks